File: Backend/Url.Shortener/Features/ShortUrls/Create/DuplicateConflictResponse.cs
namespace Url.Shortener.Features.ShortUrls.Create
{
    /// <summary>
    /// The duplicate conflict response.
    /// </summary>
    public class DuplicateConflictResponse
    {
        /// <summary>
        /// The error codename.
        /// </summary>
        public string Error => "DuplicateConflict";

        /// <summary>
        /// The error message.
        /// </summary>
        public string Message => "Failed to generate a unique short URL. Please try again later.";
    }
}

File: frontend/responses/duplicate-conflict-response.ts
interface IDuplicateConflictResponse {
    error: string;
    message: string;
};

export class DuplicateConflictResponse implements IDuplicateConflictResponse {
    error: string;
    message: string;

    constructor(error: string, message: string) {
        this.error = error;
        this.message = message;
    }
}